geode-core/src/integrationTest/java/org/apache/geode/internal/cache/xmlcache/CacheXmlGeneratorIntegrationTest.java

```diff
@@ -0,0 +1,95 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache.xmlcache;
+
+import static org.apache.geode.distributed.ConfigurationProperties.OFF_HEAP_MEMORY_SIZE;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.PartitionAttributes;
+import org.apache.geode.cache.PartitionAttributesFactory;
+import org.apache.geode.cache.Region;
+import org.apache.geode.test.junit.rules.ServerStarterRule;
+
+@RunWith(Parameterized.class)
+public class CacheXmlGeneratorIntegrationTest {
+
+  @Rule
+  public TestName testName = new TestName();
+
+  @Rule
+  public ServerStarterRule serverStarterRule = new ServerStarterRule();
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Parameterized.Parameter
+  public PartitionAttributes partitionAttributes;
+
+  @Parameterized.Parameters
+  public static Collection<PartitionAttributes> partitionAttributes() {
+    return Arrays.asList(
+        new PartitionAttributesFactory().create(),
+        new PartitionAttributesFactory().setLocalMaxMemory(16).create());
+  }
+
+  @Test
+  public void generateXmlForPartitionRegionWithOffHeapWhenDistributedSystemDoesNotExistShouldWorkProperly()
+      throws Exception {
+    CacheCreation cacheCreation = new CacheCreation();
+    RegionAttributesCreation attributes = new RegionAttributesCreation(cacheCreation);
+    attributes.setOffHeap(true);
+    attributes.setPartitionAttributes(partitionAttributes);
+    cacheCreation.createVMRegion(testName.getMethodName(), attributes);
+    File cacheXmlFile = temporaryFolder.newFile(testName.getMethodName() + ".xml");
+    PrintWriter printWriter = new PrintWriter(new FileWriter(cacheXmlFile), true);
+    CacheXmlGenerator.generate(cacheCreation, printWriter);
+
+    serverStarterRule.withProperty(OFF_HEAP_MEMORY_SIZE, "32m")
+        .withProperty("cache-xml-file", cacheXmlFile.getAbsolutePath()).startServer();
+
+    Cache cache = serverStarterRule.getCache();
+    assertThat(cache).isNotNull();
+    Region<Object, Object> region = cache.getRegion(testName.getMethodName());
+    assertThat(region).isNotNull();
+    assertThat(region.getAttributes().getOffHeap()).isTrue();
+    PartitionAttributes parsedAttributes = region.getAttributes().getPartitionAttributes();
+    assertThat(parsedAttributes.getLocalMaxMemory())
+        .isEqualTo(parsedAttributes.getLocalMaxMemory());
```

Comment:
I might be misunderstanding the purpose of these assertions, but should these assertions be comparing the values stored in "parsedAttributes" (the values we retrieve from the region) to the values in "partitionedAttributes" (the attributes initialized as a class field at the beginning of the test)? At the moment it looks like we're comparing the values in "parsedAttributes" to themselves which should always be true.

Reply:
+1 looks like it has to compare partitionAttributes and parsedAttributes

Reply:
Great catch!, thank you both!.